crates/sui-analytics-indexer/src/handlers/object_handler.rs

```diff
@@ -150,6 +150,41 @@ impl ObjectHandler {
         }
         Ok(())
     }
+
+    async fn check_type_hierarchy(
+        &self,
+        type_tag: &TypeTag,
+        package_id: ObjectID,
+        state: &mut State,
+    ) -> Result<bool> {
+        match type_tag {
+            TypeTag::Struct(s) => {
+                let object_package_id: ObjectID = s.address.into();
```

Comment:
Do you need to convert this into an `ObjectID`? It seems like you immediately convert it back into an `AccountAddress` below.

Reply:
good point, made it AccountAddress.